x-pack/plugins/snapshot_restore/server/routes/api/snapshots.ts

```diff
@@ -55,18 +116,44 @@ export function registerSnapshotsRoutes({
         return handleEsError({ error: e, response: res });
       }
 
+      // if the search is for a repository name with exact match (repository=test)
+      // and that repository doesn't exist, ES request throws an error
+      // that is why we return an empty snapshots array instead of sending an ES request
+      if (
+        searchField === 'repository' &&
+        isSearchingForNonExistentRepository(repositories, searchValue!, searchMatch, searchOperator)
+      ) {
+        return res.ok({
+          body: {
+            snapshots: [],
+            policies,
+            repositories,
+            errors: [],
+            total: 0,
+          },
+        });
+      }
       try {
         // If any of these repositories 504 they will cost the request significant time.
         const { body: fetchedSnapshots } = await clusterClient.asCurrentUser.snapshot.get({
-          repository: '_all',
-          snapshot: '_all',
+          repository:
+            searchField === 'repository'
+              ? convertSearchToWildcard(searchValue!, searchMatch, searchOperator)
+              : '_all',
           ignore_unavailable: true, // Allow request to succeed even if some snapshots are unavailable.
-          // @ts-expect-error @elastic/elasticsearch "desc" is a new param
-          order: 'desc',
-          // TODO We are temporarily hard-coding the maximum number of snapshots returned
-          // in order to prevent an unusable UI for users with large number of snapshots
-          // In the near future, this will be resolved with server-side pagination
-          size: SNAPSHOT_LIST_MAX_SIZE,
+          snapshot:
+            searchField === 'snapshot'
+              ? convertSearchToWildcard(searchValue!, searchMatch, searchOperator)
+              : '_all',
+          // @ts-expect-error @elastic/elasticsearch new API params
```

Comment:
Same here regarding an issue in https://github.com/elastic/elasticsearch-specification